src/main/java/org/opensearch/testcontainers/OpensearchContainer.java

```diff
@@ -21,10 +22,16 @@
  * (http/https) and 9300 (tcp, deprecated).
  */
 public class OpensearchContainer<SELF extends OpensearchContainer<SELF>> extends GenericContainer<SELF> {
+    // The initial password is required starting from OpenSearch 2.12.0
+    private static final Pattern OPENSEARCH_INITIAL_PASSWORD_VERSION =
+            Pattern.compile("^([3][.]\\d+[.]\\d+|[2][.][1][2-9]+[.]\\d+|[2][.][2-9]\\d+[.]\\d+|latest)$");
```

Comment:
Should we remove `3.x` reference here and just make it 2.12.0 and above?

Reply:
Not really, we need 3.0.0-SNAPSHOTs to be supported as well (plus 3.x release is approaching)

Reply:
@reta I am just saying shall we just do a regex that everything above 2.12.0 (including 3.x, 4.x, ...) instead of specifically mention 2.12+ and 3.x?

Reply:
Oh sorry,  misunderstood it. Yeah, we definitely should not support 3.x only, I will make a change shortly, thanks!